### PR TITLE
Add chunked migration using fetchmany

### PR DESCRIPTION
**Pull request by matthewoliver**
This is a much simpler version of chunked migration. And it taking
advantage of sqlalchemy's fetchmany() funtion.

### CODE REVIEW

File: psql2mysql/__init__.py [hunk 1]
```diff
@@ -167,13 +173,18 @@ def disable_constraints(self):
             "SET SESSION foreign_key_checks='OFF'"
         )
 
+    @profile
```
Comment (s-t-e-v-e-n-k) [inline psql2mysql/__init__.py]: I'd prefer if this was able to be turned off.

File: psql2mysql/__init__.py [hunk 2]
```diff
@@ -167,13 +173,18 @@ def disable_constraints(self):
...
     def writeTableRows(self, table, rows):
-        # FIXME: Allow to process this in batches instead one possibly
-        # huge transcation?
-        self.connection.execute(
-            table.insert(),
-            rows.fetchall()
-        )
+        if self.chunked and self.chunk_size > 0:
```
Comment (s-t-e-v-e-n-k) [inline psql2mysql/__init__.py]: You already check it's a positive integer in setup(), so I think just if self.chunked is fine here.

File: psql2mysql/__init__.py [hunk 3]
```diff
@@ -223,20 +243,20 @@ def migrate(self):
...
-                    table.name)
-
             # skip the schema migration related tables
             # FIXME: Should we put this into a config setting
             # (e.g. --skiptables?)
             if (table.name == "migrate_version" or
                     table.name.startswith("alembic_")):
                 continue
 
+            LOG.info("Migrating table: '%s'" % table.name)
```
Comment (s-t-e-v-e-n-k) [inline psql2mysql/__init__.py]: Why did you move this block?

File: psql2mysql/__init__.py [hunk 4]
```diff
@@ -275,6 +295,19 @@ def setupTypeDecorators(self, srcTable, targetTable):
                     targetTable.c[targetCol.name].type = decorator
 
 
+def positive_int(value):
+    try:
+        if not isinstance(value, int):
+            value = int(value)
+
+        if value < 0:
+            raise ValueError()
```
Comment (s-t-e-v-e-n-k) [inline psql2mysql/__init__.py]: Hmmm. This is almost a little dirty. Surely there is a more pythonic way ...

File: psql2mysql/__init__.py [hunk 5]
```diff
@@ -289,6 +322,15 @@ def add_subcommands(subparsers):
     parser = subparsers.add_parser(
         'migrate',
         help='Migrate data from PostgreSQL to MariaDB')
+
+    parser.add_argument("--chunked",
+                        action='store_true',
+                        default=False,
```
Comment (s-t-e-v-e-n-k) [inline psql2mysql/__init__.py]: Given the MariaDB constraints that we have about writing records, I think this should be on by default.

File: psql2mysql/__init__.py (hunk 2 above)
Comment (matthewoliver) [reply psql2mysql/__init__.py]: ahh yes. I wanted 0 to be turned off (migrate the whole table) but in debugging (ie sending none to fetchmany) actually set it to 1, so not what I wanted. Added to > 0 to force it to the else, meant to go back and clean it up.

File: psql2mysql/__init__.py (hunk 1 above)
Comment (matthewoliver) [reply psql2mysql/__init__.py]: I might remove the memory profiling. It was there while I was comparing, it's easy to add. So shouldn't be there by default. I'll remove it in the next version.

File: psql2mysql/__init__.py (hunk 3 above)
Comment (matthewoliver) [reply psql2mysql/__init__.py]: Actually I moved the skipping migrate and alembic tables up, because if were skipping them we shouldn't require the target to actually have the tables sitting there empty.

Also the empty mysql neutron db I used didn't have one of these tables (which I got from another source). so it was also causing me issues.

File: psql2mysql/__init__.py (hunk 4 above)
Comment (matthewoliver) [reply psql2mysql/__init__.py]: Happy to go rework it :) this came up while I was debugging, I should have also written a test at the same time, so bad me :)

File: psql2mysql/__init__.py (hunk 3 above)
Comment (rhafer) [reply psql2mysql/__init__.py]: Actually, we only skip those tables because it's assumed that the existing mysql database already has them, because the respecitve "db-manage" tool was already run against it. (So they should be in the very same state as the source db already). I think throwing an error if the source has them but the target doesn't is still a good idea.

File: psql2mysql/__init__.py [hunk 6]
```diff
@@ -289,6 +322,15 @@ def add_subcommands(subparsers):
     parser = subparsers.add_parser(
         'migrate',
         help='Migrate data from PostgreSQL to MariaDB')
+
+    parser.add_argument("--chunked",
```
Comment (jsuchome) [inline psql2mysql/__init__.py]: Shouldn't this be added directly into`cli_opts`

Comment (matthewoliver) [reply psql2mysql/__init__.py]: I added it here as these options only relate to the migrate sub command, if I add it at `cli_opts` they'll appear as global options.

Comment (jsuchome) [reply psql2mysql/__init__.py]: OK, I guess you are right

**Comment by jsuchome** [comment]
Could you please add a note to README file?

**Comment by matthewoliver** [comment]
I've updated, added some tests, updated the doc. And moving the option to cli_opts made some testing better. 
Is there a reason why this was closed.

**Comment by jsuchome** [comment]
Sorry for closing, I just hit the wrong button

**Comment by rhafer** [comment]
> Who can reopen this? I can't, the "reopen" button is grayed out for me cry

@nicolasbock IIRC we all that that issue. So Matt reopened it as https://github.com/SUSE/psql2mysql/pull/33 (merged meanwhile).

**Comment by nicolasbock** [comment]
@rhafer Ah, sorry I missed that. So it's all good then :smile:
